### PR TITLE
[stable/keel] keel chart

### PR DESCRIPTION
**Pull request by rimusz**
A chart that bootstraps Keel - stateless, robust and lightweight Continues Deployment system

### CODE REVIEW

File: stable/keel/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "name" . }}
```
Comment (unguiculus) [inline stable/keel/templates/deployment.yaml]: `name: {{ template "fullname" . }}`

File: stable/keel/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  annotations:
+    kubernetes.io/service-account.name: keel
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ template "name" . }}
```
Comment (unguiculus) [inline stable/keel/templates/deployment.yaml:18]: Add `release` label.

File: stable/keel/templates/secrets-webhookrelay.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.webhookRelay.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "name" . }}-webhookrelay
```
Comment (unguiculus) [inline stable/keel/templates/secrets-webhookrelay.yaml]: Should use `{{ template "fullname" . }}`. Actually, the additional suffix is not necessary. Update references accordingly.

File: stable/keel/templates/service-account.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "name" . }}
```
Comment (unguiculus) [inline stable/keel/templates/service-account.yaml]: `name: {{ template "fullname" . }}`

File: stable/keel/templates/service.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.service.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "name" . }}
```
Comment (unguiculus) [inline stable/keel/templates/service.yaml]: `name: {{ template "fullname" . }}`

File: stable/keel/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    heritage: {{ .Release.Service }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.externalPort }}
+      targetPort: 9300
+      protocol: TCP
+      name: keel
+  selector:
+    app: {{ template "name" . }}
```
Comment (unguiculus) [inline stable/keel/templates/service.yaml:20]: Add `release` label.

File: stable/keel/values.yaml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  trigger: poll
+  # polling schedule
+  pollSchedule: "@every 3m"
+  # images to track and update
+  images:
+   - repository: image.repository
+     tag: image.tag
+
+# Resources
+resources:
```
Comment (unguiculus) [inline stable/keel/values.yaml]: Don't specify defaults for resources.

File: stable/keel/Chart.yaml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+appversion: 0.4.0
+keywords:
+- kubernetes deployment
+- helm release
+- continuous deployment
+home: https://keel.sh
+sources:
+- https://github.com/rusenask/keel
+maintainers:
+- name: Rimas Mocevicius
```
Comment (unguiculus) [inline stable/keel/Chart.yaml]: Please use Github usernames.

File: stable/keel/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  email: rmocius@gmail.com
+- name: Karolis Rusenas
+  email: karolis.rusenas@gmail.com
+engine: gotpl
```
Comment (unguiculus) [inline stable/keel/Chart.yaml]: Do you have an `icon` you could add?

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR.

File: stable/keel/Chart.yaml (hunk 9 above)
Comment (rimusz) [reply stable/keel/Chart.yaml]: we do not the have the icon yet

File: stable/keel/Chart.yaml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+name: keel
+description: Open source, tool for automating Kubernetes deployment updates. Keel is stateless, robust and lightweight.
+version: 0.1.3
+appversion: 0.4.1
```
Comment (prydonius) [inline stable/keel/Chart.yaml]: Should be `appVersion`

**Review by prydonius — APPROVED** [review]
lgtm!

**Comment by prydonius** [comment]
@unguiculus lgtm, PTAL

**Comment by rimusz** [comment]
ping @unguiculus let's do not make this chart stale, please

File: stable/keel/templates/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+{{- if .Values.webhookRelay.enabled }}
+        - name: webhookrelayd
+          image: "{{ .Values.webhookRelay.image.repository }}:{{ .Values.webhookRelay.image.tag }}"
+          imagePullPolicy: {{ .Values.webhookRelay.image.pullPolicy }}
+          command: ["/webhookrelayd"]
+          env:
+            - name: KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "name" . }}-webhookrelay
```
Comment (unguiculus) [inline stable/keel/templates/deployment.yaml]: `{{ template "fullname" . }}-webhookrelay`

File: stable/keel/templates/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+                  key: key
+            - name: SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "name" . }}-webhookrelay
```
Comment (unguiculus) [inline stable/keel/templates/deployment.yaml]: `{{ template "fullname" . }}-webhookrelay`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Almost there. Secret references need to be fixed.

**Comment by rimusz** [comment]
@unguiculus I got fixed the last ones `template "fullname"`

File: stable/keel/templates/NOTES.txt [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+  NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+        Watch the status with: 'kubectl get svc --namespace kube-system -w {{ template "fullname" . }}'
+
+  export SERVICE_IP=$(kubectl get svc --namespace kube-system  {{ template "fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  echo http://$SERVICE_IP:{{ .Values.service.externalPort }}
+
+{{- else if contains "ClusterIP"  .Values.service.type }}
+
+  export POD_NAME=$(kubectl get pods --namespace kube-system  -l "app={{ template "fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (unguiculus) [inline stable/keel/templates/NOTES.txt]: `app={{ template "fullname" . }}` -> `app={{ template "name" . }},release={{ .Release.Name }}`

File: stable/keel/templates/NOTES.txt [hunk 14]
```diff
@@ -0,0 +1,33 @@
+1. The {{ template "fullname" . }} is getting provisioned in your cluster. After a few minutes, you can run the following to verify.
+
+To verify that {{ template "fullname" . }} has started, run:
+
+  kubectl --namespace=kube-system get pods -l "app={{ template "fullname" . }}"
```
Comment (unguiculus) [inline stable/keel/templates/NOTES.txt]: `app={{ template "fullname" . }}` -> `app={{ template "name" . }},release={{ .Release.Name }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/keel/templates/NOTES.txt [hunk 15]
```diff
@@ -18,7 +18,7 @@ To verify that {{ template "fullname" . }} has started, run:
 
 {{- else if contains "ClusterIP"  .Values.service.type }}
 
-  export POD_NAME=$(kubectl get pods --namespace kube-system  -l "app={{ template "fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
+  export POD_NAME=$(kubectl get pods --namespace kube-system  -l "app=app={{ template "name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (unguiculus) [inline stable/keel/templates/NOTES.txt]: One `app` too many: `app=app=`

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

**Pull request merged**